pkg/client/client.go

```diff
@@ -106,17 +106,23 @@ func (n *NutanixClientHelper) buildManagementEndpoint(ctx context.Context, nutan
 	}
 	if providerForNutanixCluster != nil {
 		providers = append(providers, providerForNutanixCluster)
+		mptr, e1 := providerForNutanixCluster.GetManagementEndpoint(envTypes.Topology{})
```

Comment:
why do we need to do it here, line 131 is already doing it rite?

Reply:
pls see link mentioned by @dkoshkin in the conversation.

Reply:
Please see this and include the unit test for the changes https://github.com/nutanix-cloud-native/cluster-api-provider-nutanix/compare/main...dkoshkin:cluster-api-provider-nutanix:dkoshkin/fix-client-always-reading-from-file?expand=1

Reply:
also pls add more unit tests if not already getting tested